File: TOUIJava/src/main/java/com/io/toui/model/Packet.java
package com.io.toui.model;

public class Packet<T> {

    //------------------------------------------------------------
    //
    public String command;

    public T data;

    public Long timestamp;

    public String id;

    //------------------------------------------------------------
    //
    public Packet() {
    }

    public Packet(
            final String _command, final T _data, final Long _timestamp, final
    String _id) {

        command = _command;
        data = _data;
        timestamp = _timestamp;
        id = _id;
    }

    @Override
    public String toString() {

        return "Packet: " + command + " : " + timestamp + " : " + id;
    }
}

File: TypeScript/src/Packet.ts
import { TOICommand } from './Command'

export class TOIPacket {
  static TERMINATOR = 0
  static ID = 16
  static TIMESTAMP = 17
  static DATA = 18

  readonly command: TOICommand
  id?: number

  constructor(command: TOICommand, id?: number) {
    this.command = command

    this.id = id
  }

  bytes(): Uint8Array {
    let result: number[] = []

    // TODO push packet id
    // TODO push packet timestamp
    // TODO push packet data

    result.push(TOIPacket.TERMINATOR)

    return new Uint8Array(result)
  }
}
